### ISSUE TITLE
Get.bottomSheet(Page()) -> does not delete Controller when closed

### ISSUE THREAD

**Issue by ValeriusGC** [jonataslaw/getx#1613]
Hello. 
When i go to new route `Get.to(() => SomePage(), binding: BindingsBuilder.put(()=>SomeController()));` or something and then do `Get.back()`, controller is deleted correct:
```
[GETX] "SomeController" onDelete() called
[GETX] "SomeController" deleted from memory
```
But in case with `Get.bottomSheet(SomePage())` with `Get.put(()=>SomeController())` inside `build()` controller remains in memory after `Get.back()`, so i have no tool to recreate controller every time i need bottomSheet. Or I have it?

**Comment by ValeriusGC** [comment]
Hi there. Of course i'm not sure it is bug, maybe it is a feature. If it's just a specific behavior, just let me know in two words.
The fact is i prefer pattern 'one page = one controller', so when Page is created or deleted I'd like to control lifecycle of its controller too. There are cases when BottomSheet's controller has to various cleaning when closed and various initialization when it creates again. `onClose, onInit, onReady` would be suit decision, I think.

**Comment by Hwan-seok** [comment]
Maybe It was fixed in 4.2.0

Ref: https://github.com/jonataslaw/getx/pull/1594

**Comment by jonataslaw** [comment]
closed by https://github.com/jonataslaw/getx/pull/1594
